tests/storage/clone.go

```diff
@@ -447,6 +450,100 @@ var _ = SIGDescribe("[Serial]VirtualMachineClone Tests", func() {
+
+				var (
+					instancetype *instancetypev1alpha2.VirtualMachineInstancetype
+					preference   *instancetypev1alpha2.VirtualMachinePreference
+				)
+
+				BeforeEach(func() {
+					snapshotStorageClass, err := getSnapshotStorageClass(virtClient)
+					Expect(err).ToNot(HaveOccurred())
+
+					if snapshotStorageClass == "" {
+						Skip("Skipping test, no VolumeSnapshot support")
+					}
+
+					instancetype = &instancetypev1alpha2.VirtualMachineInstancetype{
+						ObjectMeta: metav1.ObjectMeta{
+							GenerateName: "vm-instancetype-",
+							Namespace:    util.NamespaceTestDefault,
+						},
+						Spec: instancetypev1alpha2.VirtualMachineInstancetypeSpec{
+							CPU: instancetypev1alpha2.CPUInstancetype{
+								Guest: 1,
+							},
+							Memory: instancetypev1alpha2.MemoryInstancetype{
+								Guest: resource.MustParse("128Mi"),
+							},
+						},
+					}
+					instancetype, err := virtClient.VirtualMachineInstancetype(util.NamespaceTestDefault).Create(context.Background(), instancetype, metav1.CreateOptions{})
+					Expect(err).ToNot(HaveOccurred())
+
+					preference = &instancetypev1alpha2.VirtualMachinePreference{
+						ObjectMeta: metav1.ObjectMeta{
+							GenerateName: "vm-preference-",
+							Namespace:    util.NamespaceTestDefault,
+						},
+						Spec: instancetypev1alpha2.VirtualMachinePreferenceSpec{
+							CPU: &instancetypev1alpha2.CPUPreferences{
+								PreferredCPUTopology: instancetypev1alpha2.PreferSockets,
+							},
+						},
+					}
+					preference, err := virtClient.VirtualMachinePreference(util.NamespaceTestDefault).Create(context.Background(), preference, metav1.CreateOptions{})
+					Expect(err).ToNot(HaveOccurred())
+
+					sourceVM = tests.NewRandomVMWithDataVolumeWithRegistryImport(
+						cd.DataVolumeImportUrlForContainerDisk(cd.ContainerDiskAlpine),
+						util.NamespaceTestDefault,
+						snapshotStorageClass,
+						k8sv1.ReadWriteOnce,
+					)
+					sourceVM.Spec.Template.Spec.Domain.Resources = virtv1.ResourceRequirements{}
+					sourceVM.Spec.Instancetype = &virtv1.InstancetypeMatcher{
+						Name: instancetype.Name,
+						Kind: "VirtualMachineInstanceType",
+					}
+					sourceVM.Spec.Preference = &virtv1.PreferenceMatcher{
+						Name: preference.Name,
+						Kind: "VirtualMachinePreference",
+					}
+					sourceVM, err = virtClient.VirtualMachine(sourceVM.Namespace).Create(sourceVM)
+					Expect(err).ToNot(HaveOccurred())
+
+				})
+
+				It("should create new ControllerRevisions for cloned VM", func() {
+					By("Waiting until the source VM has instancetype and preference RevisionNames")
+					Eventually(func() bool {
+						sourceVM, err = virtClient.VirtualMachine(sourceVM.Namespace).Get(sourceVM.Name, &metav1.GetOptions{})
+						if err != nil {
+							return false
+						}
+						return sourceVM.Spec.Instancetype.RevisionName != "" && sourceVM.Spec.Preference.RevisionName != ""
+					}, 300*time.Second, 1*time.Second).Should(BeTrue())
```

Comment:
Please use description to explain this error, or else it would look like "false is expected to be true" therefore not very readable. You can simply add a description this way:
```go
 300*time.Second, 1*time.Second).Should(BeTrue(), "instance type and preference revisions are expected to not be empty")
```
Or you can use the following style which I find even better:
```go
Eventually(func() error {
	sourceVM, err = virtClient.VirtualMachine(sourceVM.Namespace).Get(sourceVM.Name, &metav1.GetOptions{})
	if err != nil {
		return false
	}
	if sourceVM.Spec.Instancetype.RevisionName != "" {
		return fmt.Errorf("instancetype revision name expected to not be empty")
	}
	
	if sourceVM.Spec.Preference.RevisionName != "" {
		return fmt.Errorf("preference revision name expected to not be empty")
	}
	
	return nil
}, 300*time.Second, 1*time.Second).ShouldNot(HaveOccured())
``

Reply:
ACK I also prefer the second style, I'll use that instead thanks!